com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -1432,6 +1489,42 @@ private void ConfigureSimulatorForUtp1()
 #endif
                 heartbeatTimeoutMS: transport.m_HeartbeatTimeoutMS);
 
+#if UTP_TRANSPORT_2_0_ABOVE
+            if (m_UseEncryption)
```

Comment:
This should also look at `m_ProtocolType` to ensure we don't configure secure parameters when using Relay.

Reply:
Can we get away with this, until we support Relay:

 ```
                 if (m_ProtocolType == ProtocolType.RelayUnityTransport)
                {
                    throw new Exception("Secure Relay transport is not supported in Netcode For GameObjects for now.");
                }
```

Reply:
I don't think we can. Unfortunately, Relay already has a (separate) way of specifying that encryption is desired (the `isSecure` parameter of `SetClientRelayData` and `SetHostRelayData`). Ideally I think we'd have a logic like this:

```csharp
if (useEncryption)
{
    if (protocolType is relay)
    {
        if (!relayServerData.IsSecure)
        {
             // log an error because we have mismatched configuration
        }
        else
        {
             // new code to support Relay+WSS (not for this PR)
        }
    }
    else
    {
        // new code this PR adds to setup encryption
    }
}
```

Reply:
That is simple enough. Fixed in 954e5f02